custom_components/waste_collection_schedule/waste_collection_schedule/source/boden_se.py

```diff
@@ -0,0 +1,106 @@
+}
+COUNTRY = "se"
+_LOGGER = logging.getLogger(__name__)
+
+# This is based on the ssam_se source, just changed to work with boden.se
+class Source:
+    def __init__(self, street_address):
+        self._street_address = street_address
+
+    def fetch(self):
+        params = {"searchText": self._street_address}
+        response = requests.post(
+            "https://edpmobile.boden.se/FutureWeb/SimpleWastePickup/SearchAdress",
+            params=params,
+            timeout=30,
+        )
+
+        address_data = json.loads(response.text)
+        address = None
+
+        if address_data and len(address_data) > 0:
+            address = address_data["Buildings"][0]
+
+        if not address:
+            return []
+
+        params = {"address": address}
+        response = requests.get(
+            "https://edpmobile.boden.se/FutureWeb/SimpleWastePickup/GetWastePickupSchedule",
+            params=params,
+            timeout=30,
+        )
+
+        data = json.loads(response.text)
+
+        entries = []
+        for item in data["RhServices"]:
+            waste_type = ""
+            next_pickup = item["NextWastePickup"]
+            try:
+                next_pickup_date = datetime.fromisoformat(next_pickup).date()
+            except ValueError as _:
+                # In some cases the date is just a month, so parse this as the
+                # first of the month to atleast get something close
+                try:
+                    next_pickup_date = datetime.strptime(next_pickup, "%b %Y").date()
+                except ValueError as month_parse_error:
+                    _LOGGER.warning(
+                        "Failed to parse date %s, %s,",
+                        next_pickup,
+                        str(month_parse_error),
+                    )
+                    continue
+
+            if item["WasteType"] == "Br\u00e4nnbart":
+                waste_type = (
+                    "Brännbart, "
+                    + item["BinType"]["ContainerType"]
+                    + " "
+                    + str(item["BinType"]["Size"])
+                    + item["BinType"]["Unit"]
+                )
+                icon = "mdi:trash-can"
+            elif item["WasteType"] == "Matavfall":
+                waste_type = (
+                    "Matavfall, "
+                    + item["BinType"]["ContainerType"]
+                    + " "
+                    + str(item["BinType"]["Size"])
+                    + item["BinType"]["Unit"]
+                )
+                icon = "mdi:trash-can"
+            else:
+                waste_type = (
+                    item["WasteType"]
+                    + " "
+                    + item["BinType"]["ContainerType"]
+                    + " "
+                    + str(item["BinType"]["Size"])
+                    + item["BinType"]["Unit"]
+                )
+                icon = "mdi:trash-can"
```

Comment:
Why are you doing the exact same thing 3 Times? The else case does exacactly the same as the if's above.
If you want to set custom icons use a ICON_MAP dictionary in the top of the file and get the icon with `ICON_MAP.get(item["WasteType"])`